packages/lib/onedrive-api.ts

```diff
@@ -343,14 +343,21 @@ export default class OneDriveApi {
 
 					await handleRequestRepeat(error);
 					continue;
-				} else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0] && !isNaN(Number(response?.headers?._headers['retry-after'][0]))) {
+				} else if (error?.code === 'activityLimitReached') {
 					// Wait for OneDrive throttling
 					// Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
 					// Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
 					i--;
-					const sleepSeconds = response.headers._headers['retry-after'][0];
+
+					const retryAfter = response.headers?.get?.('retry-after') ?? 1;
```

Comment:
```suggestion
					const retryAfter = response.headers?.get?.('retry-after') ?? '1';
```

`retryAfter` should be of type `string`.

Note that the conditionals were added here to handle the case that the `headers` API actually *doesn't* work this way for some types of `fetch` (e.g. do `fetchBlob`/`uploadBlob` always return a `Response` with `headers`?).

This should be checked.